### PR TITLE
[extractor/playsuisse] fix new url format

### PR DESCRIPTION
**Pull request by sbor23**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Support new URL scheme where episode ID is a separate unique ID.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/playsuisse.py [hunk 1]
```diff
@@ -21,10 +22,10 @@ class PlaySuisseIE(InfoExtractor):
                 'season_number': 1,
                 'episode': 'Knochen',
                 'episode_number': 1,
-                'thumbnail': 'md5:9260abe0c0ec9b69914d0a10d54c5878'
+                'thumbnail': 'md5:3cd4006d1f42cecbfdbd1ac35479c3dc'
```
Comment (pukkandan) [inline yt_dlp/extractor/playsuisse.py:25]: md5 for thumbnail is a bad idea. Can u use `re:` instead

File: yt_dlp/extractor/playsuisse.py [hunk 2]
```diff
@@ -5,10 +5,11 @@
 
 
 class PlaySuisseIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?playsuisse\.ch/watch/(?P<id>[0-9]+)'
+    _VALID_URL = r'https?://(?:www\.)?playsuisse\.ch/(watch|detail)/(.*\?episodeId=)?(?P<id>[0-9]+)?'
     _TESTS = [
         {
-            'url': 'https://www.playsuisse.ch/watch/763211/0',
```
Comment (pukkandan) [inline yt_dlp/extractor/playsuisse.py:11]: Keep the old test too

File: yt_dlp/extractor/playsuisse.py [hunk 3]
```diff
@@ -33,26 +34,60 @@ class PlaySuisseIE(InfoExtractor):
                 'title': 'Der Läufer',
                 'description': 'md5:9f61265c7e6dcc3e046137a792b275fd',
                 'duration': 5280,
-                'episode': 'Der Läufer',
-                'thumbnail': 'md5:44af7d65ee02bbba4576b131868bb783'
+                'thumbnail': 'md5:fa67fb5c1f8fae7321f71b4fbdfda617'
             }
-        },
-        {
-            'url': 'https://www.playsuisse.ch/watch/817193/0',
```
Comment (pukkandan) [inline yt_dlp/extractor/playsuisse.py:41]: Why is this removed?

File: yt_dlp/extractor/playsuisse.py [hunk 4]
```diff
@@ -5,10 +5,11 @@
 
 
 class PlaySuisseIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?playsuisse\.ch/watch/(?P<id>[0-9]+)'
+    _VALID_URL = r'https?://(?:www\.)?playsuisse\.ch/(watch|detail)/(.*\?episodeId=)?(?P<id>[0-9]+)?'
```
Comment (pukkandan) [inline yt_dlp/extractor/playsuisse.py:8]: * Avoid `.*`
* Use non-capturing groups `(?:...)`
* There can be other query
* Why did you make `id` optional?
```suggestion
    _VALID_URL = r'https?://(?:www\.)?playsuisse\.ch/(?:watch|detail)(?:[^#]*[?&]episodeId=)?(?P<id>[0-9]+)'
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/playsuisse.py (hunk 2 above)
Comment (sbor23) [reply yt_dlp/extractor/playsuisse.py:11]: Why, this format doesn't exist and work anymore?

File: yt_dlp/extractor/playsuisse.py (hunk 3 above)
Comment (sbor23) [reply yt_dlp/extractor/playsuisse.py:41]: Just because it was also an episode of a series, same as the first test. So it didn't add much value.

File: yt_dlp/extractor/playsuisse.py (hunk 4 above)
Comment (sbor23) [reply yt_dlp/extractor/playsuisse.py]: Thanks for suggestion! I had to add a slash but otherwise works fine. Optional `id` was a mistake.

File: yt_dlp/extractor/playsuisse.py (hunk 2 above)
Comment (pukkandan) [reply yt_dlp/extractor/playsuisse.py:11]: The URL still opens in browser and so should remain supported

Comment (pukkandan) [reply yt_dlp/extractor/playsuisse.py:11]: Only this is pending
